### DISCUSSION TITLE
More... -> Video qualities is too strict

### DISCUSSION THREAD

**Discussion by GrampaWildWilly** [aclap-dev/video-downloadhelper#41] [Help and Q&A]
### Your question

In the VDH settings under the More... tab there is a function called "Video qualities."  One of the options on that settings page is named "Ignore videos strictly below."  The minimum value is 240p.  I'm finding that even with it set to that, 240p variants are not being detected by VDH.  The minimum value for this should be 0, which should mean that no minimum is applied.  A value of 0 for this setting should tell VDH to detect & display all available variants regardless of their resolution.  Resolution.  Not quality.  Resolution.

The settings page you get is labelled "Media preferences."  You should make the entry in the More... list match that.  I know you're planning to remove this setting entirely.  Until you do, you should rename it from "Video qualities" to "Media preferences."

### Addon & CoApp version and details

Not that it's particularly relevant but I'm running Windows 11 Pro 64-bit version 23H2 build 22631.3085, Firefox 122.0 64-bit, licensed VDH 8.2.0.18a1 beta, CoApp 2.0.15.

**Comment by paulrouget** [comment]
If VDH can't find videos that fits within the Video Quality constrains, it will display the other qualities. So, it's not strict.

Is there a situation where you don't see a variant?

And the option "Ignore low quality videos" refers to low quality sources (not variants). For example, if VDH finds a HLS video but also a regular MP3, it won't show the MP3 as we assume the user wants to see the HLS. Unchecking that option will show the MP3.

**Reply by GrampaWildWilly** [reply]
I hit it a couple of times over here: https://github.com/aclap-dev/video-downloadhelper/discussions/31

**Reply by paulrouget** [reply]
The quality system is never "too strict". We only lower the priority of the videos which are outside of the resolution window. If there's nothing else to show, we show them.

The only situation I can think of is that a source is discarded because a better source is available. But that doesn't depend on the quality of the video, but on the origin of the video (M3U8 master origin is better than M3U8 media for example).

**Reply by GrampaWildWilly** [reply]
Well, I put the ffprobe reports & the ffmpeg logs in my posts.  You can see for the "viewer discretion advised" video @whocrazy posted, there's 4 variants in the master manifest but VDH shows only 3.  Same deal for the one @jcfrog  posted.  VDH shows 3 but there's 4 in the master manifest.  I've got "Ignore videos strictly below" set at 240p, as low as it will go, & VDH is just not showing that bottom variant.  The missing variant in both cases is 426x240.  Standard 240p is 320x240, if I'm not mistaken.  So even though the vertical dimension is exactly 240, the horizontal dimension is greater than the standard for 240p.  Maybe there's a boundary condition there that the code isn't quite handling properly.

Regardless of any issues with any specific video, I still suggest you should offer a value of 0 for that option & interpret such a value as don't do any rejecting of variants based on their resolution.  Just show everything that's in the master manifest.  Some users might be like me & want that, some might be happy to abbreviate the VDH menu & show only 1280x720 or higher.  It seems like it would be easy to satisfy everybody on this one.

**Reply by paulrouget** [reply]
> VDH shows 3 but there's 4 in the master manifest.

Is your variants count set to 3?

**Reply by GrampaWildWilly** [reply]
Oh for crying out loud . . .  It took me a while to find that.  It's not under Appearance or Behavior.  It's in the Video qualities/Media preferences.  I thought I had bumped that up.  I remember bumping that up at some point.  One of my updates must have set it back down.  OK.  That solves the mystery of the missing variant.

But my suggestion about a value of 0 still stands.  And the name on the menu should be changed to match the title on the settings page.
